### PR TITLE
Add playground project

### PR DESCRIPTION
**Pull request by egorikftp**

### CODE REVIEW

**Comment by Goooler** [comment]
Is this playground essential?

**Comment by egorikftp** [comment]
> Is this playground essential?

If I add this, it's necessary 😄 
I had a temporary project with icons and tested all changes on it. Now I'm going to use this one as "gold standard" + use it in new readme demo videos

File: playground/app/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_1_8
+        targetCompatibility = JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+    buildFeatures {
+        compose = true
+    }
```
Comment (Goooler) [inline playground/app/build.gradle.kts:38]:
```suggestion
```

Comment (Goooler) [reply playground/app/build.gradle.kts:38]: This is unnecessary from AGP 8.5.1

File: playground/app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml
Comment (Goooler) [inline playground/app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml:1]: We can just use project icon, https://github.com/ComposeGears/Valkyrie/blob/main/.idea/icon.svg

File: playground/app/src/main/res/values/colors.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <color name="purple_200">#FFBB86FC</color>
+    <color name="purple_500">#FF6200EE</color>
+    <color name="purple_700">#FF3700B3</color>
+    <color name="teal_200">#FF03DAC5</color>
+    <color name="teal_700">#FF018786</color>
+    <color name="black">#FF000000</color>
+    <color name="white">#FFFFFFFF</color>
```
Comment (Goooler) [inline playground/app/src/main/res/values/colors.xml:9]:
```suggestion
```

File: playground/gradle.properties [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# AndroidX package structure to make it clearer which packages are bundled with the
+# Android operating system, and which are packaged with your app's APK
+# https://developer.android.com/topic/libraries/support-library/androidx-rn
+android.useAndroidX=true
+# Kotlin code style for this project: "official" or "obsolete":
+kotlin.code.style=official
+# Enables namespacing of each library's R class so that its R class includes only the
+# resources declared in the library itself and none from the library's dependencies,
+# thereby reducing the size of the R class for that library
+android.nonTransitiveRClass=true
```
Comment (Goooler) [inline playground/gradle.properties:23]:
```suggestion
org.gradle.jvmargs=-Xmx2048m -Dfile.encoding=UTF-8
android.useAndroidX=true
kotlin.code.style=official
```

File: playground/local.properties
Comment (Goooler) [inline playground/local.properties:1]: This should be ignored.

File: playground/settings.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            }
+        }
+        mavenCentral()
+        gradlePluginPortal()
+    }
+}
+dependencyResolutionManagement {
+    repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)
+    repositories {
+        google()
```
Comment (Goooler) [inline playground/settings.gradle.kts:17]:
```suggestion
        google {
            content {
                includeGroupByRegex("com\\.android.*")
                includeGroupByRegex("com\\.google.*")
                includeGroupByRegex("androidx.*")
            }
        }
```

**Comment by Goooler** [comment]
> I had a temporary project with icons and tested all changes on it.

You may need test fixtures from test resources folder. If we add this sample project, you can also reuse those resources via source set addition.

**Comment by egorikftp** [comment]
@Goooler we decided to add playground as nested application module and it will be available from run configuration 🙂 

image




<hr>

> You may need test fixtures from test resources folder. If we add this sample project, you can also reuse those resources via source set addition.


I'm not sure if it's necessary at the moment since the tests contain "synthetic" icons specifically designed to check parsing and generation logic. Nonetheless, the playground is a working solution.

**Comment by Goooler** [comment]
> add playground as nested application module and it will be available from run configuration

This could be a include build, you can run the app from root project.

File: playground/app/build.gradle.kts (hunk 1 above)
Comment (egorikftp) [reply playground/app/build.gradle.kts]: done

File: playground/app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml
Comment (egorikftp) [reply playground/app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml:1]: updated icon

File: playground/app/src/main/res/values/colors.xml (hunk 2 above)
Comment (egorikftp) [reply playground/app/src/main/res/values/colors.xml]: all colors unused, removed

File: playground/gradle.properties (hunk 3 above)
Comment (egorikftp) [reply playground/gradle.properties]: reused from root project

**Review by Goooler — APPROVED** [review]

**Pull request merged**